### PR TITLE
fix: fix handling of ordered arrays when processing them async

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**

### CODE REVIEW

File: datamodel/high/base/schema.go [hunk 1]
```diff
@@ -309,38 +309,50 @@ func NewSchema(schema *base.Schema) *Schema {
...
+			go buildSchema(schemas[i], i, bChan)
 		}
 		j := 0
 		for j < totalSchemas {
 			select {
-			case t := <-bChan:
+			case r := <-bChan:
 				j++
-				*items = append(*items, t)
+				fmt.Println("got schema", r.idx, "of", totalSchemas)
```
Comment (bitomaxsp) [inline datamodel/high/base/schema.go:344]: Leftover?

Comment (TristanSpeakEasy) [reply datamodel/high/base/schema.go]: yep good catch thanks

**Comment by bitomaxsp** [comment]
Neat! thanks for fixing it this way. I can close #59 then

**Comment by TristanSpeakEasy** [comment]
> @TristanSpeakEasy When do you plan to merge this ?

As soon as I get an approval from @daveshanley

**Review by daveshanley — APPROVED** [review]
This looks great, an excellent solution, without losing the async speed! Thank you for your contribution.

**Pull request merged**